dags/utils/amplitude.py

```diff
@@ -66,6 +70,24 @@ def export_to_amplitude(
                 use_legacy_sql=False
         )
 
+        # If the underlying view requires special permissions only available to the
+        # Airflow workers, it may not be possible to maintain the view through the
+        # normal Jenkins-controlled view publishing process, so we provide a hook
+        # here to create the view with the same privileges that will be used when
+        # executing the query.
```

Comment:
@fbertsch - This is a bit of a weird compromise to handle permissions issues. I add here a `recreate_view` option that's turned on just for this email_click task. It downloads the query contents from a github url so that it can execute the `CREATE OR REPLACE VIEW` statement on the Airflow worker rather than in kube. I don't want to materialize this data to a long-lived table (which would let us sidestep this issue), as I want to keep the scope in which email clicks are available for analysis as limited as possible.

I'm definitely not in love with this solution, but I'd like to move forward with this so that I can begin testing the export in Amplitude stage. I'm open to suggestions on how better to approach this going forward.

Reply:
I'm really not sure of any good alternatives offhand. We could try to add an option to bq_etl that runs the cli with a specific service account, and provide that to the container at runtime; however that would take some time to configure and test so let's stick with this.